### PR TITLE
Add surface wetting module

### PR DESCRIPTION
**Pull request by chrisjonesBSU**
This is still a work in progress, but begins a surface wetting module that we've discussed before.  We can use this as a place to discuss the design choices we'll have to make.

### CODE REVIEW

**Comment by chrisjonesBSU** [comment]
I'm testing this out and making a list of some of the issues popping up:

- [ ] If `drop_snapshot` can be a path to a gsd file or an instance of a `Frame`, we need a check that handles the case where it is a gsd file (i.e. load with `gsd.hoomd`)
- [ ] We need to make sure `gap` and `box_height` are in the same units as the snapshot units. I'm getting an error when trying to subtract a `unit_array` with real units from a `unit_array` with dimensionless units

File: flowermd/modules/surface_wetting/surface_wetting.py [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+        # shift drop particles z position to be at the top of surface
+        z_shift = (
+            np.abs(min(drop_pos[:, 2]) - max(surface_pos[:, 2])) - self.gap
+        )
+        drop_pos[:, 2] -= z_shift
+        wetting_pos = np.concatenate((surface_pos, drop_pos), axis=0)
+        return wetting_pos
+
+    @property
+    def wetting_snapshot(self):
```
Comment (chrisjonesBSU) [inline flowermd/modules/surface_wetting/surface_wetting.py:336]: For consistency purposes, I think we should call this property `hoomd_snapshot`. That is what we use in `System` and what I was using as an attribute in the welding module.

**Review by chrisjonesBSU — COMMENTED** [review]
Left a couple notes, let me know what you think!

**Comment by chrisjonesBSU** [comment]
The tutorial looks good! Just one suggestion, I think we should try to use the same formatting we have in pre-commit in the tutorial code snippets.

For example:

```
graphene = Graphene(x_repeat=20, y_repeat=20, n_layers=1,
                    base_units=droplet_system.reference_values)
graphene.apply_forcefield(force_field=OPLS_AA(), remove_charges=True,
                          auto_scale=False, r_cut=2.5)
```

Should become

```
graphene = Graphene(
    x_repeat=20,
    y_repeat=20,
    n_layers=1,
    base_units=droplet_system.reference_values
)

graphene.apply_forcefield(
    force_field=OPLS_AA(),
    remove_charges=True,
    auto_scale=False,
    r_cut=2.5
)
```

I think it is easier to read and following along this way.

**Review by marjanAlbouye — APPROVED** [review]

**Pull request merged**